src/main/java/com/code_intelligence/jazzer/mutation/mutator/lang/EnumMutatorFactory.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2023 Code Intelligence GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.code_intelligence.jazzer.mutation.mutator.lang;
+
+import static com.code_intelligence.jazzer.mutation.combinator.MutatorCombinators.mutateIntegralInRange;
+import static com.code_intelligence.jazzer.mutation.combinator.MutatorCombinators.mutateThenMapToImmutable;
+import static com.code_intelligence.jazzer.mutation.support.TypeSupport.findFirstParentIfClass;
+
+import com.code_intelligence.jazzer.mutation.api.MutatorFactory;
+import com.code_intelligence.jazzer.mutation.api.SerializingMutator;
+import java.lang.reflect.AnnotatedType;
+import java.util.Optional;
+
+final class EnumMutatorFactory extends MutatorFactory {
+  @Override
+  public Optional<SerializingMutator<?>> tryCreate(AnnotatedType type, MutatorFactory factory) {
+    return findFirstParentIfClass(type, Enum.class).map(parent -> {
+      Enum[] values = ((Class<Enum>) type.getType()).getEnumConstants();
+      // TODO: when 1 or 0 values, return a constant mutator?
+      if (values.length == 0) {
+        return null;
+      }
+      return mutateThenMapToImmutable(mutateIntegralInRange(0, values.length - 1),
+          (index) -> values[index], Enum::ordinal, "Enum");
```

Comment:
At the moment it's not working (it prints ´Enum´ and not `EnumTest` in the tests).

Reply:
True, it uses the declared return type of the lambda, not the actual class of the returned value. The latter could be achieved by constructing an element with something like `readExclusive(new ByteArrayInputStream(new byte[0]))` and then calling `getClass` on that.

@bertschneider What do you think, should we mention concrete types in the debug string? My initial thought was not to do this so that debug strings are certain not to contain user information (e.g. concrete class names), but of course that additional information could still be useful.

Reply:
IMHO concrete type names, even user defined ones, are fine, as they are not published anywhere and can help greatly during debugging.